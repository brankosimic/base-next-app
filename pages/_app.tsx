import type { AppProps } from 'next/app'
import React from 'react'
import { StoreProvider } from '../contexts'
import '../styles/default.scss'

const MyApp = ({ Component, pageProps }: AppProps): JSX.Element => (
  <StoreProvider {...pageProps}>
    <Component {...pageProps} />
  </StoreProvider>
)

export default MyApp
