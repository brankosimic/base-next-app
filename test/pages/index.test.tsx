import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

const H2 = 'h2-description'
const DESC = 'some description'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home seoDynamicDescription={DESC} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('H2 rendered correctly', () => {
    const { getByTestId } = render(<Home seoDynamicDescription={DESC} />, {})
    expect(getByTestId(H2).textContent).toBe(DESC)
  })
})
