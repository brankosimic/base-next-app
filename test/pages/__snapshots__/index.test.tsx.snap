// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1808515695 container"
  >
    <main
      class="jsx-1808515695"
    >
      <h3 />
      <div
        class="MuiFormControl-root MuiTextField-root myInput"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="standard-basic"
          id="standard-basic-label"
        >
          Enter name
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="standard-basic"
            type="text"
            value=""
          />
        </div>
      </div>
      <h1
        class="jsx-1808515695 title"
      >
        Welcome 
      </h1>
      <h2
        class="jsx-1808515695"
        data-testid="h2-description"
      >
        some description
      </h2>
    </main>
    <footer
      class="jsx-1808515695"
    >
      <a
        class="jsx-1808515695"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iMzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Vercel Logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
